nipype/pipeline/engine/workflows.py

```diff
@@ -770,16 +770,46 @@ def _check_nodes(self, nodes):
     def _has_attr(self, parameter, subtype="in"):
         """Checks if a parameter is available as an input or output
         """
+        hierarchy = parameter.split(".")
+        attrname = hierarchy.pop()
+        nodename = hierarchy.pop()
+
+        targetworkflow = self
+        for workflowname in hierarchy:
+            workflow = None
+            for node in targetworkflow._graph.nodes():
+                if node.name == workflowname:
+                    if isinstance(node, Workflow):
+                        workflow = node
+                        break
+            if workflow is None:
+                return False
+            targetworkflow = workflow
+
+        targetnode = None
+        for node in targetworkflow._graph.nodes():
+            if node.name == nodename:
+                if isinstance(node, Workflow):
+                    return False
+                else:
+                    targetnode = node
+                    break
+        if targetnode is None:
+            return False
+
         if subtype == "in":
-            subobject = self.inputs
+            if not hasattr(node.inputs, attrname):
```

Comment:
targetnode?

Reply:
Yes, that should be targetnode. 
```suggestion
            if not hasattr(targetnode.inputs, attrname):
```